feat(github):Add workflow for running unittests for pull requests

Wishmaster-de opened the pull request:
ToDos:

- [x] Change URL for badge or switch branch  name
- [x] check return value of 'nproc'

Wishmaster-de committed:
feat(github):Add workflow for running unittests for pull requests

Wishmaster-de committed:
feat(doc): Add unittest execution state badge

DanielJerolm commented:
Looks great!
I think I'll extend GPCC's CMake build files by a target "unittest executable", which is build if GPCC is "stand-alone". That would allow you to get rid off gpcc_dev and toolchain_configs in your build_test.yaml file.

Wishmaster-de committed:
feat(github): add files to ignore list which should not trigger the unittest workflow

- to minimize the execution time on GitHub do not trigger the execution of the unittest workflow for changes on specific files

DanielJerolm commented on .github/workflows/build-test.yaml:
```diff
@@ -0,0 +1,43 @@
+      run: ./cmake_config_nativelinux-unittest-debug.sh
+
+    - name: Build
+      run: cmake --build ${{github.workspace}}/build_unittest --config ${{env.BUILD_TYPE}} --parallel 4
```
Would `./built_unitttest.sh all` also work?

Wishmaster-de replied:
> Would `./built_unitttest.sh all` also work?

I'll give it a try.
I need to check which number of cores 'nproc' returns. https://github.com/Wishmaster-de/gpcc/blob/e93a85518cab8296291bb382e17c2becbce95e91/scripts/build_x.sh#L52
If the number is greater than 4 i need to adjust the number of cores to 4. Otherwise the workflow will be canceled because i am not a paying user.  ;-)

Wishmaster-de replied:
> Would `./built_unitttest.sh all` also work?

Yes it works. No i am using the script. :-)
